miqa/core/models/project.py

```diff
@@ -97,22 +97,23 @@ def get_status(self):
         tier_2_reviewers = [
             user.id for user in get_users_with_perms(self, only_with_perms_in=['tier_2_reviewer'])
         ]
-        scans_in_project = (
-            Scan.objects.filter(experiment__in=self.experiments.all())
-            .prefetch_related('decisions')
-            .annotate(decision_count=models.Count('decisions'))
-        )
-        complete_scans_in_project = [
-            scan
-            for (scan, last_decision) in [
-                (scan, ScanDecision.objects.filter(scan=scan).order_by('created')[0])
-                for scan in scans_in_project.filter(decision_count__gt=0)
-            ]
-            if last_decision.decision == 'U' or last_decision.creator.id in tier_2_reviewers
-        ]
+        scans_in_project = Scan.objects.filter(experiment__project=self)
+        completed_scans_in_project = scans_in_project.alias(
+            latest_decider_id=models.Subquery(
+                ScanDecision.objects.filter(scan__id=models.OuterRef('id'))
+                .order_by('-created')
+                .values('creator_id')[:1]
+            ),
+            latest_decision=models.Subquery(
+                ScanDecision.objects.filter(scan__id=models.OuterRef('id'))
+                .order_by('-created')
+                .values('decision')[:1]
+            ),
+        ).filter(models.Q(latest_decider_id__in=tier_2_reviewers) | models.Q(latest_decision='U'))
```

Comment:
This is a somewhat complicated query, but it emits the SQL that I was intending:

```sql
SELECT COUNT(*)
  FROM (
        SELECT "core_scan"."id" AS Col1
          FROM "core_scan"
         INNER JOIN "core_experiment"
            ON ("core_scan"."experiment_id" = "core_experiment"."id")
         WHERE ("core_experiment"."project_id" = '323e798a-3d2b-4d4e-9ab7-cde1206b5bbb'::uuid AND ((SELECT U0."creator_id" FROM "core_scandecision" U0 WHERE U0."scan_id" = "core_scan"."id" ORDER BY U0."created" DESC LIMIT 1) IN (1) OR (SELECT U0."decision" FROM "core_scandecision" U0 WHERE U0."scan_id" = "core_scan"."id" ORDER BY U0."created" DESC LIMIT 1) = 'U'))
       )
```

Note that I was also able to simplify the `scans_in_project` query to get rid of the annotations that required aggregation.

Because this is a complex (perhaps tending towards inscrutable) query, we should definitely write a unit test for this `get_status` method. The test should use our factories to create a Project with several Scans and ScanDecisions, and assert that the `get_status` counts come back correctly. I leave the creation of that test to you @annehaley 
😄

Reply:
Test added in [d4a8a0a](https://github.com/OpenImaging/miqa/pull/567/commits/d4a8a0a5c07c29a1e00ba6d9998d090fd6e97524)